public/app/features/query/components/QueryGroup.tsx

```diff
@@ -94,6 +97,28 @@ export class QueryGroup extends PureComponent<Props, State> {
       next: (data: PanelData) => this.onPanelDataUpdate(data),
     });
 
+    if (config.featureToggles.enableDatagridEditing) {
+      const dashboard = getDashboardSrv().getCurrent();
+      if (dashboard) {
+        this.datagridEditEventSubscription = dashboard.events.getStream(DatagridDataChangeEvent).subscribe({
```

Comment:
What is this subscription for ?

Reply:
Think we need another way for panels to update query . This feels like the wrong pace / way

Reply:
No really sure where to set this. The subscription watches for datagrid data changes. Whenever we edit the datagrid panel the snapshot is published to update the dashboard model.

Reply:
@mdvictor yes, but this cannot exist here. something specific to the DataGridPanel cannot live in the global group group component. This needs to be handled from the panel.

Reply:
I think this could be implemented with a new optional method on PanelContext, like onUpdateQueries?: (queries: DataQuery) => void, and then implemented in PanelStateWrapper. 

what do you think @ryantxu @dprokop ?

Reply:
The main problem is that QueryGroup would not update it's state when the optional method would be called on the PanelContext. I've committed some code to address that. Also, instead of adding a new method on the PanelContext I've found that publishing the `PanelQueriesChangedEvent` event updates the `PanelEditorTabs`.